src/views/CreateDandisetView/CreateDandisetView.vue

```diff
@@ -42,29 +70,57 @@
 <script lang="ts">
 import { defineComponent, computed, ref } from '@vue/composition-api';
 import { dandiRest, loggedIn } from '@/rest';
+import { IdentifierForAnAward } from '@/types/schema';
+
+// Regular expression to validate an NIH award number.
+// Based on https://era.nih.gov/files/Deciphering_NIH_Application.pdf
+// and https://era.nih.gov/erahelp/commons/Commons/understandGrantNums.htm
+const NIH_AWARD_REGEX = /^\d \w+ \w{2} \d{6}-\d{2}([A|S|X|P]\d)?$/;
+
+function awardNumberValidator(awardNumber: IdentifierForAnAward): boolean {
+  return NIH_AWARD_REGEX.test(awardNumber);
+}
+
+const VALIDATION_FAIL_MESSAGE = 'Award number must be properly space-delimited.\n\nExample (exclude quotes):\n"1 R01 CA 123456-01A1"';
 
 export default defineComponent({
   name: 'CreateDandisetView',
   setup(props, ctx) {
     const name = ref('');
     const description = ref('');
-    const saveDisabled = computed(() => !name.value || !description.value);
+    const embargoed = ref(false);
+    const awardNumber = ref('');
+    const saveDisabled = computed(
+      () => !name.value
+      || !description.value
+      || (embargoed.value && !awardNumberValidator(awardNumber.value)),
+    );
+
+    const awardNumberRules = computed(
+      () => [(v: string) => awardNumberValidator(v) || VALIDATION_FAIL_MESSAGE],
```

Comment:
Is there any reason not to just define `awardNumberValidator` as `return NIH_AWARD_REGEX.test(awardNumber) || VALIDATION_FAIL_MESSAGE` so you don't need this wrapper?
Also, I don't see anything that would necessitate this being a computed property. I don't actually know if you can just return normal objects from `setup` or if it has to wrapped in a `ref`.

Reply:
> Also, I don't see anything that would necessitate this being a computed property. I don't actually know if you can just return normal objects from `setup` or if it has to wrapped in a `ref`.

The docs for `v-text-field` specify it as a computed property. If `computed` is removed validation no longer works.

> Is there any reason not to just define `awardNumberValidator` as `return NIH_AWARD_REGEX.test(awardNumber) || VALIDATION_FAIL_MESSAGE` so you don't need this wrapper?

Yes, because both the `awardNumberRules` and `saveDisabled` computed vars need to use it.